core/server/master/src/test/java/alluxio/master/block/ConcurrentBlockMasterTest.java

```diff
@@ -417,7 +423,7 @@ public void concurrentCommitWithSameWorkerHeartbeatSameBlock() throws Exception
           // The same block is removed on worker in this heartbeat
           // This should succeed as commit locks the block exclusively and finishes first
           // When the block heartbeat processes the same block, it has been committed
-          Command cmd = mBlockMaster.workerHeartbeat(worker1,
+          Command cmd = mBlockMaster.workerHeartbeat(worker1, CLUSTER_ID,
```

Comment:
Should we consider concurrent tests? Like when the worker registers with master, there are concurrent read/write requests? The concurrency happens in two sides:
1. There are concurrent `removeBlock(blockId, worker)` requests on the BlockMaster with worker register
2. There are concurrent `writeBlock()` requests on the BlockWorker with worker register.

I think there can be too many combinations of events so we can never enumerate them all. But we can start with reasoning what operations can be concurrent and if there could be race conditions. If we identify a race condition, try to translate that into a unit test.